### ISSUE TITLE
CTE reference misinterpreted as recursive CTE

### ISSUE THREAD

**Issue by ricardoekm** [duckdb/duckdb#9187]
### What happens?

I'm migrating a system from Postgres to Duck, the following simplified query works in Postgres but fails in DuckDB. For some reason it's interpreting that as a recursive CTE.

```sql
CREATE TABLE people (
    name text
);

WITH people AS (SELECT * FROM people),
people_cte AS (SELECT * FROM people)
SELECT * FROM people_cte
```

Error
```
java.sql.SQLException: Binder Error: Circular reference to CTE \"people\", use WITH RECURSIVE to use recursive CTEs;
```

### To Reproduce

Just run the above query.

### OS:

Apple Silicon - ARM

### DuckDB Version:

0.8.1

### DuckDB Client:

JDBC

### Full Name:

Ricardo Mayerhofer

### Affiliation:

Hopara

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by szarnyasg** [comment]
@ricardoekm Thanks for reporting!

When reproducing via CLI, I get a more detailed error message with two possible suggestions:

```sql
D WITH people AS (SELECT * FROM people),
> people_cte AS (SELECT * FROM people)
> SELECT * FROM people_cte
> ;
Error: Binder Error: Circular reference to CTE "people", There are two possible solutions.
1. use WITH RECURSIVE to use recursive CTEs.
2. If you want to use the TABLE name "people" the same as the CTE name, please explicitly add "SCHEMA" before table name. You can try "main.people" (main is the duckdb default schema)
```

That said, let me cc @kryonix on this issue who may have more to say.

**Comment by ricardoekm** [comment]
Thanks @szarnyasg! An additional input, the following query seems to work fine

```sql
WITH people AS (SELECT * FROM people)
SELECT * FROM people
```

**Comment by ricardoekm** [comment]
Adding the schema "main." to the table name yields the same error.

**Comment by l1t1** [comment]
`main.people`  works in version 0.9
```
root@localhost:~# ./duckdb
v0.9.0 0d84ccf478
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
D CREATE TABLE people (
>     name text
> );
D
D WITH people AS (SELECT * FROM people),
> people_cte AS (SELECT * FROM people)
> SELECT * FROM people_cte;
Error: Binder Error: Circular reference to CTE "people", There are two possible solutions.
1. use WITH RECURSIVE to use recursive CTEs.
2. If you want to use the TABLE name "people" the same as the CTE name, please explicitly add "SCHEMA" before table name. You can try "main.people" (main is the duckdb default schema)
D WITH people AS (SELECT * FROM main.people), people_cte AS (SELECT * FROM people) SELECT * FROM people_cte;
┌─────────┐
│  name   │
│ varchar │
├─────────┤
│ 0 rows  │
└─────────┘
```

**Comment by l1t1** [comment]
`main.people`  works in version 0.9
```
root@localhost:~# ./duckdb
v0.9.0 0d84ccf478
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
D CREATE TABLE people (
>     name text
> );
D
D WITH people AS (SELECT * FROM people),
> people_cte AS (SELECT * FROM people)
> SELECT * FROM people_cte;
Error: Binder Error: Circular reference to CTE "people", There are two possible solutions.
1. use WITH RECURSIVE to use recursive CTEs.
2. If you want to use the TABLE name "people" the same as the CTE name, please explicitly add "SCHEMA" before table name. You can try "main.people" (main is the duckdb default schema)
D WITH people AS (SELECT * FROM main.people), people_cte AS (SELECT * FROM people) SELECT * FROM people_cte;
┌─────────┐
│  name   │
│ varchar │
├─────────┤
│ 0 rows  │
└─────────┘
```

**Comment by kryonix** [comment]
This is a binder issue. The CTE name `people` takes precedence over the table `people`, instead of using the table as a fallback like PostgreSQL would do. As already mentioned, `main.people` fixes this for version 0.9. But maybe we can do a better job handling such cases during the binding phase. I will have a look at it.

**Comment by szarnyasg** [comment]
@kryonix thanks!
